packages/rocketchat-plugin-handler/server/server.js

```diff
@@ -0,0 +1,72 @@
+
+export const plugin_handler ={};
+plugin_handler.plugins = [];
+function remove_user_from_automatic_channel(user, plugins) {
+	const channelNames = plugins.map(function(x) { return x.channelName(user); });
+
+	const userSubscriptions = RocketChat.models.Subscriptions.findByTypeAndUserId('c', user.user._id).fetch();
+	userSubscriptions.forEach((arrayItem) => {
+		if (!channelNames.includes(arrayItem.name) && arrayItem._room.automatic) {
+		// Remove the user from this other channel.
+			const room = RocketChat.models.Rooms.findOneById(arrayItem._room._id);
+			RocketChat.removeUserFromRoom(room._id, user.user);
+
+			//delete the user if it is last.(There may be a race condition)
+			if (room.usernames.length===1) {
+				Meteor.call('eraseRoom', room._id, true);
+
+			}
+		}
+	});
+}
+
+export function leave_automatic_channel(user, room, plugins) {
+	//plugins is an array which has the names of those channels for which admin wants the blacklisted feature to work
+	if (plugins.includes(room.plugin_name)) {
+		RocketChat.models.Users.update({ _id: user._id }, { $addToSet: { ignored_automatic_channels: room.name } });
+		if (room.usernames.length===1) {
```

Comment:
Please use spaces between operators and values -> `if (room.usernames.length === 1) {`